code/missionui/missionbrief.cpp

```diff
@@ -1787,6 +1870,110 @@ void brief_do_frame(float frametime)
 	}
 }
 
+// -------------------------------------------------------------------------------------
+// brief_api_do_frame()
+//
+// this is a pared down version of brief_do_frame() for use with the Scpui API - Mjn
+// frametime is in seconds
+//
+void brief_api_do_frame(float frametime)
+{
+
+	// This may be needed by a future PR to get map icon clicking working through the API - Mjn
+	// if (Closeup_icon) {
+	//	Brief_mouse_up_flag = 0;
+	//}
+	//gr_reset_clip();
+
+	if (!Background_playing) {
+		int time = -1;
+		int check_jump_flag = 1;
+
+		if (Current_brief_stage != Last_brief_stage) {
+
+			// Check if we have a quick transition pending
+			if (Quick_transition_stage != -1) {
+				Quick_transition_stage = -1;
+				brief_reset_last_new_stage();
+				time = 0;
+				check_jump_flag = 0;
+			}
+
+			if (check_jump_flag) {
+				if (abs(Current_brief_stage - Last_brief_stage) > 1) {
+					Quick_transition_stage = Current_brief_stage;
+					Current_brief_stage = Last_brief_stage;
+					Assert(Current_brief_stage >= 0);
+					Start_fade_up_anim = 1;
+					goto Transition_done;
```

Comment:
Oof, this is some old and crusty code.
I get that you don't want to rework all of that though, and since it's a mirror of the non-API function, I guess it's fine.

Reply:
Yes, for now rewriting to avoid goto is outside the scope and the API brief map should work exactly as the normal one does.